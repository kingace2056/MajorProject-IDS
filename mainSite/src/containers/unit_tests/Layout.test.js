import React from "react";
import ReactDOM from "react-dom";
import { createStore, compose, applyMiddleware, combineReducers } from "redux";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import authReducer from "../../store/reducers/auth";
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from "react-router-dom";
import BaseRouter from "../../routes";

import CustomLayout from "../Layout"

const composeEnhances = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const rootReducer = combineReducers({
    auth: authReducer
});

const store = createStore(rootReducer, composeEnhances(applyMiddleware(thunk)));

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Provider store={store}>
        <Router>
            <CustomLayout>
                <BaseRouter />
            </CustomLayout>
        </Router>
    </Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<Provider store={store}>
            <Router>
                <CustomLayout>
                    <BaseRouter />
                </CustomLayout>
            </Router>
        </Provider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});