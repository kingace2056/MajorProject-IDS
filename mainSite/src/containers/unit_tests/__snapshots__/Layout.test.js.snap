// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="ui inverted top fixed menu"
  >
    <div
      className="ui container"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="header item"
          id="menu_item_home"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="home inverted icon"
            onClick={[Function]}
          />
           Home
        </div>
      </a>
      <a
        href="/login"
        onClick={[Function]}
      >
        <div
          className="header item"
          id="menu_item_login"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="sign-in inverted icon"
            onClick={[Function]}
          />
           Login
        </div>
      </a>
      <a
        href="/signup"
        onClick={[Function]}
      >
        <div
          className="header item"
          id="menu_item_signup"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="signup inverted icon"
            onClick={[Function]}
          />
           Signup
        </div>
      </a>
    </div>
  </div>
  <div>
    <div>
      <div />
      <div
        className="ui vertical segment"
        style={
          Object {
            "padding": "8em 0em",
          }
        }
      >
        <div
          className="ui container stackable middle aligned grid"
        >
          <div
            className="row"
          >
            <div
              className="eight wide column"
            >
              <h3
                className="ui header"
                style={
                  Object {
                    "color": "black",
                    "fontSize": "3em",
                  }
                }
              >
                Intrusion Detection System
              </h3>
              <p
                style={
                  Object {
                    "color": "#3d3b3b",
                    "fontSize": "1.5em",
                  }
                }
              >
                An intrusion detection system is one such system that can be a building block in making the IOT network as secure as possible, that passively monitors the data exchange in the network and of the network with externals entities and looks for malicious activities that can be classified as an intrusion or attack.
              </p>
              <h3
                className="ui header"
                style={
                  Object {
                    "color": "black",
                    "fontSize": "3em",
                  }
                }
              >
                We care about Protection
              </h3>
              <p
                style={
                  Object {
                    "color": "#3d3b3b",
                    "fontSize": "1.5em",
                  }
                }
              >
                Includes IOT networks like CCTVs, smoke detectors, fire alarms, speaker systems, thermostats, vending machines, etc.
              </p>
            </div>
            <div
              className="right floated six wide column"
            >
              <img
                alt="ids"
                size="large"
                src="IDS.webp"
              />
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="center aligned column"
            >
              <br />
              <br />
              <br />
              <h3
                className="ui header"
                style={
                  Object {
                    "color": "black",
                    "fontSize": "2em",
                  }
                }
              >
                CHECK OUT THE ATTACKS DETECTED
              </h3>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ui vertical segment"
        style={
          Object {
            "padding": "0em",
          }
        }
      >
        <div
          className="ui stackable internally celled equal width grid"
        >
          <div
            className="center aligned row"
          >
            <div
              className="column"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "paddingBottom": "2em",
                  "paddingTop": "2em",
                }
              }
            >
              <div
                align="center"
                className="ui card"
                onClick={[Function]}
              >
                <img
                  alt="ddos"
                  height="200"
                  src="ddos.jpg"
                />
                <div
                  className="content"
                >
                  <div
                    className="header"
                  >
                    DDOS ATTACK
                  </div>
                  <div
                    className="description"
                  >
                    A distributed denial-of-service (DDoS) attack is a malicious attempt to disrupt normal traffic of a targeted server, service or network by overwhelming the target or its surrounding infrastructure with a flood of Internet traffic
                  </div>
                </div>
              </div>
            </div>
            <div
              className="column"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "paddingBottom": "2em",
                  "paddingTop": "2em",
                }
              }
            >
              <div
                className="ui card"
                onClick={[Function]}
              >
                <img
                  alt="dtype"
                  height="200"
                  src="dtype.jpg"
                />
                <div
                  className="content"
                >
                  <div
                    className="header"
                  >
                    DATA TYPE PROBING ATTACK
                  </div>
                  <div
                    className="description"
                  >
                    Probing is a type of attack in which the intruder scans network devices to determine weakness in topology design or some opened ports and then use them in the future for illegal access to personal information                 
                  </div>
                </div>
              </div>
            </div>
            <div
              className="column"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "paddingBottom": "2em",
                  "paddingTop": "2em",
                }
              }
            >
              <div
                className="ui card"
                onClick={[Function]}
              >
                <img
                  alt="scan"
                  height="200"
                  src="scan.jpg"
                />
                <div
                  className="content"
                >
                  <div
                    className="header"
                  >
                    SCAN ATTACK
                  </div>
                  <div
                    className="description"
                  >
                    A port scan is an attack that sends client requests to a range of server port addresses on a host, with the goal of finding an active port and exploiting a known vulnerability of that service
                  </div>
                </div>
              </div>
            </div>
            <div
              className="column"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "paddingBottom": "2em",
                  "paddingTop": "2em",
                }
              }
            >
              <div
                className="ui card"
                onClick={[Function]}
              >
                <img
                  alt="wrong"
                  height="200"
                  src="wrong.jpeg"
                />
                <div
                  className="content"
                >
                  <div
                    className="header"
                  >
                    WRONG SETUP ATTACK
                  </div>
                  <div
                    className="description"
                  >
                    It is a type of attack where attackers attempt to remove the connection or manipulate the connection of the IOT network
                  </div>
                </div>
              </div>
            </div>
            <div
              className="column"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "paddingBottom": "2em",
                  "paddingTop": "2em",
                }
              }
            >
              <div
                className="ui card"
                onClick={[Function]}
              >
                <img
                  alt="mitm"
                  height="200"
                  src="mitm.jpeg"
                />
                <div
                  className="content"
                >
                  <div
                    className="header"
                  >
                    MAN IN THE MIDDLE ATTACK
                  </div>
                  <div
                    className="description"
                  >
                    It is an attack where the attacker secretly relays and possibly alters the communications between two parties who believe that they are directly communicating with each other
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ui vertical segment"
        style={
          Object {
            "padding": "8em 0em",
          }
        }
      >
        <div
          className="ui text container"
        >
          <h3
            className="ui header"
            style={
              Object {
                "color": "black",
                "fontSize": "2em",
              }
            }
          >
            Overall working of the System
          </h3>
          <p
            style={
              Object {
                "color": "#3d3b3b",
                "fontSize": "1.33em",
              }
            }
          >
            This IDS is designed to run in the background of an IoT network, where distance data is processed and sent from the node (combination of ultrasonic sensor, Arduino, Nodemcu) to the apache server. The network traffic is monitored using Tshark and dumped into a csv file. The python script in the backend reads this csv file, preprocesses it and feeds it to the ML model. If an intrusion is detected, it is notified to the user on the front end. The ML model Random Forest is used as its accuracy was proven to be better than any other model that was tested. The model was trained using the data collected from the IoT network traffic.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui inverted vertical segment"
    style={
      Object {
        "margin": "5em 0em 0em",
        "padding": "5em 0em",
      }
    }
  >
    <div
      className="ui center aligned container"
    >
      <div
        className="ui inverted stackable divided grid"
      >
        <div
          className="three wide column"
        >
          <h4
            className="ui inverted header"
          >
            Group Members
          </h4>
          <div
            className="ui inverted link list"
            role="list"
          >
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              Shruti Houji
            </a>
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              Rhishabh Hattarki
            </a>
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              Sahil Dixit
            </a>
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              Sanika Patil
            </a>
          </div>
        </div>
        <div
          className="three wide column"
        >
          <h4
            className="ui inverted header"
          >
            PRN Number
          </h4>
          <div
            className="ui inverted link list"
            role="list"
          >
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              71707035C
            </a>
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              71707027B
            </a>
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              71706924K
            </a>
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              71707451L
            </a>
          </div>
        </div>
        <div
          className="three wide column"
        >
          <h4
            className="ui inverted header"
          >
            Guide
          </h4>
          <div
            className="ui inverted link list"
            role="list"
          >
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              M. R. Dhage
            </a>
          </div>
        </div>
        <div
          className="seven wide column"
        >
          <h4
            className="ui inverted header"
            style={
              Object {
                "paddingTop": "3em",
              }
            }
          >
            Anomaly Based Intrusion Detection System For IoT Networks Using Random Forest
          </h4>
        </div>
      </div>
      <div
        className="ui inverted section divider"
      />
      <div
        className="ui small divided horizontal inverted link list"
        role="list"
      >
        <a
          className="item"
          href="https://github.com/s3r-be/be-project"
          id="github_link"
          onClick={[Function]}
          role="listitem"
          target="_blank"
        >
          Click here to check the code on Github
        </a>
      </div>
    </div>
  </div>
</div>
`;
